.evergreen/config.yml

```diff
@@ -641,7 +641,18 @@ tasks:
         - func: "set aws temp creds"
         - func: "run tests"
           vars:
+            TESTS: "csfle"
+
+    - name: "test-csfle-mongocryptd"
+      commands:
+        - func: "bootstrap mongo-orchestration"
+          vars:
+            TOPOLOGY: "replica_set"
             SKIP_CRYPT_SHARED: "yes"
```

Comment:
> I'm not sure if the previous tests properly tested without crypt_shared. After adding the initial commit to output some basic information about the test environment, I noticed that crypt_shared was still available. The SKIP_CRYPT_SHARED variable was incorrectly set in the call to "run tests", but needs to be set when starting mongo-orchestration as that function downloads crypt_shared unless asked not to do so.

`SKIP_CRYPT_SHARED` was introduced to `run-tests.sh` in [0fe1a0f](https://github.com/mongodb/mongo-php-library/commit/0fe1a0f0b9ee54ce270980b2651b94fcc58921ff#diff-3954529b61794d5a9a1f941f18a65a3233f21167980e973154ba1aa8324901a5R11) for [PHPLIB-911](https://jira.mongodb.org/browse/PHPLIB-911), but the environment variable was never actually used in that script. Instead `CRYPT_SHARED_LIB_PATH` is always exported.

![](https://media.giphy.com/media/RiNhy3LdnA12yxTwQe/giphy.gif)

I think my original intention for PHPLIB-911 was to only export `CRYPT_SHARED_LIB_PATH` to PHPUnit if `SKIP_CRYPT_SHARED=no` (default). But that would only have worked if the downloaded copy of crypt_shared never ends up in a default system path. `run-tests.sh` ultimately has no control over whether one of the download scripts places crypt_shared in a system path (in which case `CRYPT_SHARED_LIB_PATH` isn't even relevant).

This suggests that `SKIP_CRYPT_SHARED` isn't relevant to `run-tests.sh` at all and perhaps is only needed for `download-mongodb.sh` and `download-crypt.sh`, which are called via `run-orchestration.sh` and `serverless/create-instance.sh`, respectively. For context, see [this search for `CRYPT_SHARED_LIB_PATH` in drivers-evergreen-tools](https://github.com/search?q=repo%3Amongodb-labs%2Fdrivers-evergreen-tools%20CRYPT_SHARED_LIB_PATH&type=code).

I propose we remove `SKIP_CRYPT_SHARED` from `run-tests.sh`. `CRYPT_SHARED_LIB_PATH` should remain.

----

Side note: AFAIK, there's no mechanism to completely disable loading of crypt_shared (see: [CSFLE: extraOptions](https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#extraoptions)). This may be related to some extra testing-specific options some drivers have (see: [this Slack thread in #drivers-fle](https://mongodb.slack.com/archives/CFPHLTRMK/p1689084650285559)), none of which would be available to PHP anyway as they're private libmongoc APIs.

Reply:
FWIW, that's the reason for the configuration output I added to bootstrap.php. I wanted to see the configuration to make sure it does what we expect, that's how I noticed the above.

I've removed `SKIP_CRYPT_SHARED` from run-tests.sh and its invocations.